catalogue_pipeline/ingestor/src/test/scala/uk/ac/wellcome/platform/ingestor/services/WorkIndexerTest.scala

```diff
@@ -108,6 +90,100 @@ class WorkIndexerTest
     }
   }
 
+  describe("updating merged / redirected works") {
+    it(
+      "doesn't override a merged Work with same version but merged flag = false") {
+      val mergedWork = createIdentifiedWorkWith(version = 3, merged = true)
+      val unmergedWork = mergedWork.copy(merged = false)
+
+      withLocalElasticsearchIndex { indexName =>
+        withWorkIndexer { workIndexer =>
+          val unmergedWorkInsertFuture = for {
```

Comment:
I've refactored these tests cause there was a lot of repetition. I prefer to keep the "action" (can't come up with a better word) and the assertion on the action result separate in general, though. 
I think squishing them together often makes less visible what is being tested and leads to assert functions with very vague names that make that even less easy to understand. 
So in this case, I've chosen to retain some of the code duplication because I don't want to hide what's the sequence works being ingested. 
I was thinking about moving that for comprehension into a `ingestWorksInOrder` or something like that though